tfx/orchestration/experimental/kubernetes/kubernetes_dag_runner.py

```diff
@@ -0,0 +1,378 @@
+      if kubernetes_component_launcher.KubernetesComponentLauncher.can_launch(
+          component.executor_spec, component_config):
+        LaunchAsContainerComponent(component,
+                                   component_launcher_class,
+                                   component_config,
+                                   pipeline).run_component()
+      else:
+        wrapped_component = _wrap_container_component(
+            component=component,
+            component_launcher_class=component_launcher_class,
+            component_config=component_config,
+            pipeline=pipeline
+        )
+
+        # reload properties
+        (wrapped_component_launcher_class,
+         wrapped_component_config) = config_utils.find_component_launch_info(
+             self._config, wrapped_component)
+
+        LaunchAsContainerComponent(wrapped_component,
+                                   wrapped_component_launcher_class,
+                                   wrapped_component_config,
+                                   pipeline).run_component()
+
+      ran_components.add(component)
+
+
+  def _run_as_kubernetes_job(self, pipeline: tfx_pipeline.Pipeline) -> None:
+    """Submits and runs a tfx pipeline from outside the cluster.
+
+    Args:
+      pipeline: Logical pipeline containing pipeline args and components.
+    """
+    serialized_pipeline = self._serialize_pipeline(pipeline)
+    arguments = [
+        '--serialized_pipeline',
+        serialized_pipeline,
+    ]
+    batch_api = kube_utils.make_batch_v1_api()
+    job_name = 'Job_' + pipeline.pipeline_info.run_id
+    pod_label = _sanitize_pod_name(job_name)
+    container_name = 'pipeline-orchestrator'
+    job = kube_utils.make_job_object(
+        name=job_name,
+        container_image=_TFX_IMAGE,
+        command=_DRIVER_COMMAND + arguments,
+        container_name=container_name,
+        pod_labels={
+            'job-name': pod_label,
+        },
+    )
+    try:
+      batch_api.create_namespaced_job(
+          "default", job, pretty=True)
+    except client.rest.ApiException as e:
+      raise RuntimeError('Failed to submit job! \nReason: %s\nBody: %s' %
+                         (e.reason, e.body))
+
+    # Wait for pod to start
+    orchestrator_pods = []
+    core_api = kube_utils.make_core_v1_api()
+    start_time = datetime.datetime.utcnow()
+    while not orchestrator_pods and (datetime.datetime.utcnow() - start_time
+                                     ).seconds < 300:
+      try:
+        orchestrator_pods = core_api.list_namespaced_pod(
+            namespace='default',
+            label_selector='job-name={}'.format(pod_label)
+        ).items
+      except client.rest.ApiException as e:
+        if e.status != 404:
+          raise RuntimeError('Unknown error! \nReason: %s\nBody: %s' %
+                             (e.reason, e.body))
+      time.sleep(1)
+
+    # transient orchestrator should only have 1 pod
```

Comment:
(In the separate PR): Can we do a consistency check here?  Also formatting for the comment.

Reply:
Thanks, added the check.